<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1600</width>
    <height>720</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1000</width>
    <height>400</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1600</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset theme="iot_icon">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame_total">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>10</y>
      <width>1581</width>
      <height>701</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(229, 229, 229);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QFrame" name="frame_header">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>1581</width>
       <height>75</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(41, 42, 55);</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QLabel" name="lbl_logo">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>15</y>
        <width>441</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Black</family>
        <pointsize>24</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color:#FFF;</string>
      </property>
      <property name="text">
       <string>EMS IoT DASHBOARD</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame_logo">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>5</y>
        <width>64</width>
        <height>64</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-image: url(:/hmi/iot_icon);
background-repeat: no-repeat;</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="frame_greeting">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>300</width>
       <height>561</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(35,36,54);
border-style: outset;
border-color:#0000FF;
border-width:5px;
border-radius:20px;
</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QFrame" name="frame_weather_logo">
      <property name="geometry">
       <rect>
        <x>118</x>
        <y>216</y>
        <width>64</width>
        <height>64</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-image: url(:/hmi/sun_icon);
background-repeat: no-repeat;
border:none;</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_greeting">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>298</y>
        <width>271</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI</family>
        <pointsize>20</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">border:none;
color:#FFF;
</string>
      </property>
      <property name="text">
       <string>Good Morning!</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_status_1">
      <property name="geometry">
       <rect>
        <x>35</x>
        <y>356</y>
        <width>231</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI</family>
        <pointsize>12</pointsize>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">border:none;
color:#FFF;
background-color:none;</string>
      </property>
      <property name="text">
       <string>Today's weather is good~</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_date">
      <property name="geometry">
       <rect>
        <x>45</x>
        <y>30</y>
        <width>211</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI</family>
        <pointsize>20</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(255, 170, 0);
border:none;
</string>
      </property>
      <property name="text">
       <string>2022-06-01</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_day">
      <property name="geometry">
       <rect>
        <x>45</x>
        <y>82</y>
        <width>211</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI</family>
        <pointsize>20</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(255, 170, 0);
border:none;
</string>
      </property>
      <property name="text">
       <string>Wednesday</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_time">
      <property name="geometry">
       <rect>
        <x>45</x>
        <y>139</y>
        <width>211</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI</family>
        <pointsize>26</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(255, 170, 0);
border:none;
</string>
      </property>
      <property name="text">
       <string>22:07</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_status_2">
      <property name="geometry">
       <rect>
        <x>35</x>
        <y>387</y>
        <width>231</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI</family>
        <pointsize>12</pointsize>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">border:none;
color:#FFF;
background-color:none;</string>
      </property>
      <property name="text">
       <string>Today's weather is good~</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_status_3">
      <property name="geometry">
       <rect>
        <x>35</x>
        <y>416</y>
        <width>231</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI</family>
        <pointsize>12</pointsize>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">border:none;
color:#FFF;
background-color:none;</string>
      </property>
      <property name="text">
       <string>Today's weather is good~</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="frame_temp">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>90</y>
       <width>350</width>
       <height>561</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(35,36,54);
border-style: outset;
border-color:#0000FF;
border-width:5px;
border-radius:20px;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QLabel" name="lbl_temp_head">
      <property name="geometry">
       <rect>
        <x>101</x>
        <y>16</y>
        <width>231</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Historic</family>
        <pointsize>14</pointsize>
        <weight>99</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:none;
font: 75 14pt &quot;Segoe UI Historic&quot;;
border:none;
font-weight:800;
color: rgb(255, 255, 255);
</string>
      </property>
      <property name="text">
       <string>Global Temperature</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QFrame" name="frame_temp_logo">
      <property name="geometry">
       <rect>
        <x>14</x>
        <y>60</y>
        <width>128</width>
        <height>128</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-image: url(:/hmi/temp_128);
background-repeat: no-repeat;
border:none;</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_temp_val">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>90</y>
        <width>131</width>
        <height>101</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Historic</family>
        <pointsize>72</pointsize>
        <weight>99</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(255, 127, 39);
border :none;
background-color:none;
font: 72pt &quot;Segoe UI Historic&quot;;
font-weight:900;</string>
      </property>
      <property name="text">
       <string>22</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_celcius">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>90</y>
        <width>41</width>
        <height>101</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Historic</family>
        <pointsize>72</pointsize>
        <weight>99</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(255, 127, 39);
border :none;
background-color:none;
font: 75 72pt &quot;Segoe UI Historic&quot;;
font-weight:900;</string>
      </property>
      <property name="text">
       <string>°</string>
      </property>
     </widget>
     <widget class="QDial" name="dial_temp">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>240</y>
        <width>291</width>
        <height>291</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="maximum">
       <number>60</number>
      </property>
      <property name="singleStep">
       <number>5</number>
      </property>
      <property name="pageStep">
       <number>5</number>
      </property>
      <property name="value">
       <number>5</number>
      </property>
      <property name="sliderPosition">
       <number>5</number>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="invertedControls">
       <bool>true</bool>
      </property>
      <property name="wrapping">
       <bool>false</bool>
      </property>
      <property name="notchTarget">
       <double>0.050000000000000</double>
      </property>
      <property name="notchesVisible">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="frame_humid">
     <property name="geometry">
      <rect>
       <x>690</x>
       <y>90</y>
       <width>350</width>
       <height>561</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(35,36,54);
border-style: outset;
border-color:#0000FF;
border-width:5px;
border-radius:20px;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QFrame" name="frame_humid_logo">
      <property name="geometry">
       <rect>
        <x>14</x>
        <y>60</y>
        <width>128</width>
        <height>128</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-image: url(:/hmi/humid_128);
background-repeat: no-repeat;
border:none;</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_humid_head">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>16</y>
        <width>231</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Historic</family>
        <pointsize>14</pointsize>
        <weight>99</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:none;
font: 75 14pt &quot;Segoe UI Historic&quot;;
border:none;
font-weight:800;
color: rgb(255, 255, 255);
</string>
      </property>
      <property name="text">
       <string>Global Humidity</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_humid_val">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>90</y>
        <width>131</width>
        <height>101</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Historic</family>
        <pointsize>72</pointsize>
        <weight>99</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(0, 162, 232);
border :none;
background-color:none;
font: 72pt &quot;Segoe UI Historic&quot;;
font-weight:900;</string>
      </property>
      <property name="text">
       <string>54</string>
      </property>
     </widget>
     <widget class="QLabel" name="lbl_humid_per">
      <property name="geometry">
       <rect>
        <x>283</x>
        <y>130</y>
        <width>61</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Historic</family>
        <pointsize>40</pointsize>
        <weight>99</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(0, 162, 232);
border :none;
background-color:none;
font: 40pt &quot;Segoe UI Historic&quot;;
font-weight:900;</string>
      </property>
      <property name="text">
       <string>%</string>
      </property>
     </widget>
     <widget class="QDial" name="dial_humid">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>240</y>
        <width>291</width>
        <height>291</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="maximum">
       <number>60</number>
      </property>
      <property name="singleStep">
       <number>5</number>
      </property>
      <property name="pageStep">
       <number>5</number>
      </property>
      <property name="value">
       <number>5</number>
      </property>
      <property name="sliderPosition">
       <number>5</number>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="invertedControls">
       <bool>true</bool>
      </property>
      <property name="wrapping">
       <bool>false</bool>
      </property>
      <property name="notchTarget">
       <double>0.050000000000000</double>
      </property>
      <property name="notchesVisible">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="frame_history">
     <property name="geometry">
      <rect>
       <x>1050</x>
       <y>90</y>
       <width>511</width>
       <height>561</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(35,36,54);
border-style: outset;
border-color:#0000FF;
border-width:5px;
border-radius:20px;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QLabel" name="lbl_humid_head_2">
      <property name="geometry">
       <rect>
        <x>261</x>
        <y>16</y>
        <width>231</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Historic</family>
        <pointsize>14</pointsize>
        <weight>99</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:none;
font: 75 14pt &quot;Segoe UI Historic&quot;;
border:none;
font-weight:800;
color: rgb(255, 255, 255);
</string>
      </property>
      <property name="text">
       <string>Data History</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QFrame" name="frame_chart">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>471</width>
        <height>471</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(230,236,234);
border-color:#000000;
border-width:0px;
border-radius:10px;</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QTextEdit" name="txtResult">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>431</width>
         <height>411</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QLabel" name="lbl_mqtt_stat">
     <property name="geometry">
      <rect>
       <x>65</x>
       <y>670</y>
       <width>301</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>나눔고딕</family>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>STATUS</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnStart">
     <property name="geometry">
      <rect>
       <x>22</x>
       <y>662</y>
       <width>30</width>
       <height>30</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 0, 0);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources>
  <include location="hmi_window.qrc"/>
 </resources>
 <connections/>
</ui>
